diff -urNp fwupd-1.0.1.old/data/remotes.d/lvfs.conf fwupd-1.0.1/data/remotes.d/lvfs.conf
--- fwupd-1.0.1.old/data/remotes.d/lvfs.conf	2017-11-09 16:07:29.596004147 +0000
+++ fwupd-1.0.1/data/remotes.d/lvfs.conf	2017-11-09 16:07:46.905897123 +0000
@@ -1,7 +1,7 @@
 [fwupd Remote]
 
 # this remote provides metadata and firmware marked as 'stable' from the LVFS
-Enabled=true
+Enabled=false
 Title=Linux Vendor Firmware Service
 Keyring=gpg
 MetadataURI=https://s3.amazonaws.com/lvfsbucket/downloads/firmware.xml.gz
