diff -urNp fwupd-0.8.2.old/data/fwupd.service.in fwupd-0.8.2/data/fwupd.service.in
--- fwupd-0.8.2.old/data/fwupd.service.in	2017-07-03 14:30:09.361974001 +0100
+++ fwupd-0.8.2/data/fwupd.service.in	2017-07-03 14:30:32.740828053 +0100
@@ -8,12 +8,6 @@ Before=gdm.service
 Type=dbus
 BusName=org.freedesktop.fwupd
 ExecStart=@servicedir@/fwupd/fwupd
-MemoryDenyWriteExecute=yes
 PrivateTmp=yes
-ProtectControlGroups=yes
 ProtectHome=yes
-ProtectKernelModules=yes
-ProtectSystem=full
 RestrictAddressFamilies=AF_NETLINK AF_UNIX
-RestrictRealtime=yes
-ReadWritePaths=@localstatedir@/lib/fwupd -@bootdir@
